clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1161,6 +1172,14 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
   Args.AddLastArg(CmdArgs, options::OPT_C);
   Args.AddLastArg(CmdArgs, options::OPT_CC);
 
+  // When generating the using the integration footer, add the comments
+  // to the first preprocessing step.
+  if (!Args.hasArg(options::OPT_C) && JA.isOffloading(Action::OFK_SYCL) &&
```

Comment:
> !Args.hasArg(options::OPT_C)

What would go wrong if the option _was_ present in the user's command? Is it some CC1 warning on duplicate arguments that we're trying to avoid?